pkg/controller/plan/validation_test.go

```diff
@@ -0,0 +1,56 @@
+package plan
+
+import (
+	"github.com/konveyor/forklift-controller/pkg/controller/base"
+	ginkgo "github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/version"
+	discovery "k8s.io/client-go/discovery/fake"
+	"k8s.io/client-go/kubernetes/fake"
+)
+
+var _ = ginkgo.Describe("Plan Validations", func() {
+	var (
+		fakeClientSet *fake.Clientset
+		reconciler    *Reconciler
+	)
+
+	ginkgo.BeforeEach(func() {
+		reconciler = &Reconciler{
+			base.Reconciler{},
+		}
+		fakeClientSet = fake.NewSimpleClientset()
+	})
+
+	ginkgo.Describe("validateOCPVersion", func() {
+		ginkgo.Context("when the OpenShift version is supported", func() {
+			ginkgo.BeforeEach(func() {
+				fakeDiscovery, ok := fakeClientSet.Discovery().(*discovery.FakeDiscovery)
+				gomega.Expect(ok).To(gomega.BeTrue())
+				fakeDiscovery.FakedServerVersion = &version.Info{
+					Major: "1", Minor: "26",
+				}
+			})
```

Comment:
do we really need the before each on each case? is it possible to pull most of it out to the global before each? on each case maybe just set the fakeversion?